src/pages/ConfirmOperationPage/index.tsx

```diff
@@ -0,0 +1,164 @@
+            </div>
+            <div className="confirm-operation__description">
+              Please enter 2FA code to confirm withdraw:
+            </div>
+            <div className="confirm-operation-form">
+              <div
+                className={classnames('form-group', {
+                  'has-error': !!this.error || this.isTimeout
+                })}
+              >
+                <label htmlFor="code" className="control-label">
+                  2FA Code
+                </label>
+                <div className="error-bar" />
+                <TextMask
+                  className="form-control"
+                  mask={[/\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/]}
+                  name="code"
+                  id="code"
+                  guide={false}
+                  // tslint:disable-next-line:jsx-no-lambda
+                  onChange={(e: any) =>
+                    (this.code = e.currentTarget.value.replace(' ', ''))}
+                  value={this.code}
+                />
+                {this.error && <span className="help-block">{this.error}</span>}
+                {this.isTimeout && (
+                  <span className="help-block">
+                    Operation timeout.{' '}
+                    <a href="#" onClick={this.handleTryAgain}>
+                      Try again
+                    </a>
+                  </span>
+                )}
+              </div>
+              <div className="confirm-operation-form__actions">
+                <input
+                  type="submit"
+                  value="Confirm"
+                  className="btn btn--primary"
+                  onClick={this.handleSubmit}
+                  disabled={this.isLoading || this.isTimeout || !this.code}
+                />
+                {this.isLoading && <Spinner />}
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  private handleTryAgain = async (e: any) => {
+    e.preventDefault();
+    const operationId = await this.withdrawStore.sendWithdrawCryptoRequest(
+      this.withdrawStore.withdrawCrypto.assetId,
+      this.withdrawStore.withdrawCrypto
+    );
+    this.props.history.replace(ROUTE_CONFIRM_OPERATION_ID(operationId));
+  };
+
+  private handleSubmit = async () => {
+    const {operationId} = this.props.match.params;
+    this.isLoading = true;
+    this.error = '';
+    const actions = {
+      [OpStatus.ConfirmationRequested]: () => {
+        this.error = 'Code is not valid';
+        this.isLoading = false;
+      },
+      [OpStatus.Completed]: () => {
+        this.props.history.replace(ROUTE_WITHDRAW_CRYPTO_SUCCESS);
+      },
+      [OpStatus.Confirmed]: () => {
+        this.props.history.replace(ROUTE_WITHDRAW_CRYPTO_SUCCESS);
+      },
+      [OpStatus.Failed]: () => {
+        this.props.history.replace(ROUTE_WITHDRAW_CRYPTO_FAIL);
+      }
+    };
+
+    const TIMEOUT_LIMIT = 10000;
+    const timeout = window.setTimeout(async () => {
```

Comment:
Name is too common, I think it make sense to rename this handle variable into something more specific